### PR TITLE
fix: TRIAG-1864: Export snapshot filenames should use the custom snap…

### PR DESCRIPTION
**Pull request by luarmr**
For now this is me and **Alec** playing around, don't suffer for this yet

# Use Custom Titles in Export Snapshot Filenames

## Summary

This PR enables export snapshots to use custom titles in their filenames instead of the default `project-{id}-at-{timestamp}-{md5}` pattern. When a user provides a custom snapshot name, it will be sanitized and used as the filename prefix, making exported files more identifiable and user-friendly.

### Backend Changes

1. **Added `sanitize_filename()` helper function** (`data_export/mixins.py`)
   - Sanitizes custom titles for safe use in filenames
   - Removes invalid filename characters
   - Replaces spaces with hyphens
   - Limits length to 200 characters
   - Returns `None` if title is empty/invalid (triggers fallback to default pattern)

2. **Updated `save_file()` method** (`data_export/mixins.py`)
   - Uses sanitized custom title if available
   - Falls back to default pattern if no title or sanitization fails
   - Format: `{sanitized-title}-{md5[0:8]}.json` or `project-{id}-at-{timestamp}-{md5[0:8]}.json`

3. **Updated `async_convert()` function** (`data_export/api.py`)
   - Uses sanitized title for converted format files (CSV, TSV, COCO, etc.)
   - Same fallback behavior as main export files

4. **Fixed download endpoint filename extraction** (`data_export/api.py`)
   - Uses `os.path.basename()` to extract just the filename from the full storage path
   - Prevents full path (`export/{project_id}/{filename}`) from appearing in download headers
   - Fixes issue where filenames included path components like `export_198575_...`

## Example

**Before:**
- Filename: `project-198575-at-2026-01-23-16-16-48a6bac9.json`
- Download header: `export_198575_project-198575-at-2026-01-23-16-16-48a6bac9`

**After (with custom title "Alec is cool"):**
- Filename: `Alec-is-cool-48a6bac9.json`
- Download header: `Alec-is-cool-48a6bac9.json`

**After (without custom title):**
- Filename: `project-198575-at-2026-01-23-16-16-48a6bac9.json` (unchanged)
- Download header: `project-198575-at-2026-01-23-16-16-48a6bac9.json` (fixed)

### ✅ Safe Areas (No Breaking Changes)

- **Backward Compatibility**: Exports without custom titles continue to use the default pattern
- **File Storage**: Directory structure unchanged (`export/{project_id}/{filename}`)
- **Database**: No schema changes (title field already existed)
- **Frontend**: No changes needed (already sends title in API requests)
- **Deprecated Sync Export**: Separate code path, unaffected

### ⚠️ Areas Requiring Attention

- **Deprecated Endpoints**: 
  - `ProjectExportFilesAuthCheck` and `ProjectExportFiles` parse filenames using `split('-')[0]` to extract project ID
  - These endpoints are deprecated (`exclude=True` in OpenAPI) and likely not used in production
  - May not work correctly with custom titles, but this is acceptable given their deprecated status

### Manual Testing Checklist

- [x] Create export with custom title → Filename uses custom title
- [x] Create export without title → Filename uses default pattern
- [x] Create export with special characters → Characters sanitized correctly
- [x] Create export with very long title → Title truncated appropriately
- [x] Convert export to CSV/TSV → Converted files use custom title
- [x] Download export → Headers contain only filename (not full path)
- [x] Download old exports → Still works correctly
- [x] Export list displays correctly → Both old and new exports shown

### Test Plan

See [EXPORT_FILENAME_CUSTOM_TITLE_TEST_PLAN.md](./EXPORT_FILENAME_CUSTOM_TITLE_TEST_PLAN.md) for comprehensive test cases.

## Filename Sanitization Rules

The `sanitize_filename()` function:
- Removes invalid filename characters (`/`, `\`, `:`, `*`, `?`, `"`, `<`, `>`, `|`)
- Replaces spaces and multiple hyphens with single hyphens
- Removes leading/trailing hyphens
- Limits length to 200 characters
- Returns `None` if result is empty (triggers default pattern)

**Examples:**
- `"My Export"` → `"My-Export"`
- `"Test/File:Name?"` → `"TestFileName"`
- `"  Multiple   Spaces  "` → `"Multiple-Spaces"`
- `"---Test---"` → `"Test"`
- `"!!!@@@###"` → `None` (falls back to default)

## Files Changed

```
label_studio/data_export/mixins.py
  - Added sanitize_filename() function
  - Updated save_file() method to use custom titles

label_studio/data_export/api.py
  - Updated async_convert() to use custom titles
  - Fixed download endpoint to use basename() for filename extraction
```

## Migration Notes

- **No migration required**: Changes are backward compatible
- **Existing exports**: Continue to work with old filename pattern
- **New exports**: Automatically use custom titles when provided
- **Rollback**: Safe to revert; existing exports unaffected

## Related Issues

- Fixes issue where custom snapshot titles were not reflected in downloaded filenames
- Fixes issue where download filenames included full storage path

### Before
```
Downloaded filename: export_198575_project-198575-at-2026-01-23-16-16-48a6bac9.json
```

### After (with custom title "Alec is cool")
```
Downloaded filename: Alec-is-cool-48a6bac9.json
```

## Checklist

- [x] Code follows project style guidelines
- [x] Changes are backward compatible
- [x] No database migrations required
- [x] No breaking API changes
- [x] Test plan documented
- [x] Edge cases handled (empty titles, special characters, etc.)
- [x] Fallback to default pattern when title unavailable
- [x] Download headers fixed to use basename only

## Future Improvements

- Consider updating deprecated endpoints to handle custom titles
- Add validation for title length in frontend
- Consider allowing users to see sanitized filename before export creation
- Update documentation to mention custom title support

### CODE REVIEW

File: label_studio/data_export/api.py [hunk 1]
```diff
@@ -591,16 +591,20 @@ def get(self, request, *args, **kwargs):
                 # below header tells NGINX to catch it and serve, see docker-config/nginx-app.conf
                 redirect = '/file_download/' + protocol + '/' + url.replace(protocol + '://', '')
                 response['X-Accel-Redirect'] = redirect
-                response['Content-Disposition'] = 'attachment; filename="{}"'.format(file.name)
-                response['filename'] = os.path.basename(file.name)
+                # Use basename to get just the filename, not the full path
+                filename = os.path.basename(file.name)
+                response['Content-Disposition'] = 'attachment; filename="{}"'.format(filename)
+                response['filename'] = filename
```
Comment (luarmr) [inline label_studio/data_export/api.py:597]: Current are like export_{project_id}_project-{project_id}  this will remove that, so this is a change in behavior

**Comment by robot-ci-heartex** [comment]
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

**Comment by robot-ci-heartex** [comment]
This PR was closed because it has been stalled for 10 days with no activity.
